libraries/AP_Networking/AP_Networking_port.cpp

```diff
@@ -368,22 +368,26 @@ bool AP_Networking::Port::send_receive(void)
             WITH_SEMAPHORE(sem);
             n = writebuffer->peekbytes(buf, available);
         }
-        if (n > 0) {
-            const auto ret = sock->send(buf, n);
+        if (n > 0 && strlen(last_udp_connect_address) != 0 && last_udp_connect_port != 0) {
+            const auto ret = sock->sendto(buf, n, last_udp_connect_address, last_udp_connect_port);
             if (ret > 0) {
                 WITH_SEMAPHORE(sem);
                 writebuffer->advance(ret);
                 active = true;
             }
         }
-    } else {
-        if (type == NetworkPortType::UDP_SERVER && have_received) {
-            // connect the socket to the last receive address if we have one
-            char buf[16];
-            uint16_t last_port;
-            const char *last_addr = sock->last_recv_address(buf, sizeof(buf), last_port);
-            if (last_addr != nullptr && port != 0) {
-                connected = sock->connect(last_addr, last_port);
+    }
+    if (type == NetworkPortType::UDP_SERVER && have_received) {
+        // connect the socket to the last receive address if we have one
+        char buf[16] {};
+        uint16_t last_port = 0;
+        const char *last_addr = sock->last_recv_address(buf, sizeof(buf), last_port);
```

Comment:
add a last_recv_address() varient that takes a uint32_t &addr reference in host byte order, and a sendto() that takes a uint32_t address, so we are not doing the string translation on every pkt